src/CommonProviderImplementation/Helpers.fs

```diff
@@ -208,6 +210,7 @@ module internal ProviderHelpers =
                   ResolutionFolder = resolutionFolder }
             
             let readText() = 
+                lazy(
```

Comment:
So actually pressing the "(" key with the inner-most MyXmlNode will cause XMLProvider to execute all the code four times?